src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,70 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { useAppDispatch } from '../../app/hooks';
+import { actions } from '../../features/currentTodo';
+import { Todo } from '../../types/Todo';
+
+type TodoModalProps = {
+  currentTodo: Todo;
+};
+
+export const TodoModal: React.FC<TodoModalProps> = ({
+  currentTodo: {
+    userId, title, id, completed,
+  },
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    getUser(userId).then(setUser);
+  }, []);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
+      {!user ? (
+        <Loader />
+      ) : (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              {`Todo #${id}`}
+            </div>
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
-
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={() => dispatch(actions.removeTodo())}
```

Comment:
you can extract this function to the top to be:

`const removeTodo = () => dispatch(actions.removeTodo());`

and use here just as a callback:

`onClick={removeTodo}`